### ISSUE TITLE
Scenes corrupted when switching to "Import As" Animation Library. 

### ISSUE THREAD

**Issue by nubbyninja** [godotengine/godot#81671]
### Godot version

Version: v4.1.1.stable.mono.official [bd6af8e0e]

### System information

Win 11 - Forward+ 

### Issue description

This is my first Godot project, coming from Unity, so please bear with me.

Tested with multiple .glb files 

Models are exported form Blender 3.3.10 but also tested 3.3.4 and had same issue.

### Steps to reproduce

On the Import Panel,
Import As: Animation Library
Reimport
-editor closes and reopens-
"Scene corrupted"
Scene is no long accessible/usable.
![image](https://github.com/godotengine/godot/assets/59494766/935fcb41-5350-448e-a29b-848eb55d182e)

### Minimal reproduction project

Jump to \Project\Models\Characters
[Depart(Godot).zip](https://github.com/godotengine/godot/files/12614108/Depart.Godot.zip)

**Comment by Calinou** [comment]
I think this is "expected", in the sense that your inherited scene depends on a resource that doesn't exist anymore once you change the GLB's import type:

```
[ext_resource type="PackedScene" uid="uid://bbf56uaq1yg05" path="res://Project/Models/Characters/greyboxman.glb" id="1_bj302"]
```

This results in one of the scene's dependencies not being found when you open the scene, and Godot currently doesn't feature a way to resolve these issues from the editor.

Godot should probably warn when changing an import type of a resource that is depended upon in other scenes/resources, so that you can prepare for those situations in advance. Missing dependency resolution is also something that should be added: https://github.com/godotengine/godot-proposals/issues/6838

**Comment by nubbyninja** [comment]
So this isn't as much a bug, as a missing implementation for warnings / prompts. 
"You're about to orphan a scene or scenes that depend on this asset. Converting this asset to "Animation Library" will effectively break those scenes. Are you sure?"

I did warn that I was very new >_<

I am willing to test with other versions of Blender if you think that will help. Blender Launcher makes that fairly easy.

**Comment by mrhebisan** [comment]
I would disagree that this is a matter of a missing prompt.  You shouldn't be able to break entire chains of scenes so easily, in such a way that the editor is unable to open the broken scenes.  In playing with this I broke three whole scenes because of the fact that the leaf scene was corrupted and deleting that scene caused the scenes in which it was instanced to fail to open with an unhelpful error message.  I did eventually go in manually and delete the offending references in a text editor to salvage my scenes, but this would be a terrible issue with deep nesting.

Instead of failing to open with an unhelpful error message it would be better if the editor could open the scene anyway but with some indication that a node is broken (perhaps red text, or a strikethrough).  If other corruption is detected then at the least it would be ideal to point us in the file to where the corruption is.
